content/influxdb/v2.0/query-data/common-queries/count_unique_values_for_column.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Find and count unique values
+description: >
+  Count the number of unique values in a specified column.
+influxdb/v2.0/tags: [queries]
+menu:
+  influxdb_2_0:
+    name: Count unique values
+    parent: Common queries
+weight: 104
+---
+
+The following examples use the Sample NOAA weather data to identify and count unique locations data was collected from.
```

Comment:
This comment applies to other places as well – We have a "Sample Data" page with information pointing to other sample data sets. We should add the NOAA weather data set there and link to it from these pages.

https://docs.influxdata.com/influxdb/v2.0/reference/sample-data/

On the sample data page, we just need to describe what the data set is and how people can use it. So for this dataset, we'll want to include a Flux script that uses the experimental CSV package to pull the data from s3.

**Note:** This dataset is about 10MB and each time they execute the script, it will re-download the entire dataset. We may recommend that they create a new bucket (`noaa`), then run the following to write the data to the new bucket. That way they can query the data locally without having to re-download all of it with each execution. If we go this route, we should also update the query to pull from a local bucket rather than pulling csv from a remote URL (I'd prefer this anyway).

```js
import "experimental/csv"

csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
  |> to(bucket: "noaa", org: "your-org")
```

Reply:
Excellent suggestion, thank you @sanderson!